instrumentation/opentelemetry-instrumentation-openai/src/opentelemetry/instrumentation/openai/patch.py

```diff
@@ -0,0 +1,438 @@
+        llm_prompts = []
+        for item in kwargs.get("messages", []):
+            tools = get_tool_calls(item)
+            if tools is not None:
+                tool_calls = []
+                for tool_call in tools:
+                    tool_call_dict = {
+                        "id": tool_call.id if hasattr(tool_call, "id") else "",
+                        "type": tool_call.type if hasattr(tool_call, "type") else "",
+                    }
+                    if hasattr(tool_call, "function"):
+                        tool_call_dict["function"] = {
+                            "name": (
+                                tool_call.function.name
+                                if hasattr(tool_call.function, "name")
+                                else ""
+                            ),
+                            "arguments": (
+                                tool_call.function.arguments
+                                if hasattr(tool_call.function, "arguments")
+                                else ""
+                            ),
+                        }
+                    tool_calls.append(tool_call_dict)
+                llm_prompts.append(tool_calls)
+            else:
+                llm_prompts.append(item)
+
+        span_attributes = {
+            **get_llm_request_attributes(kwargs, prompts=llm_prompts),
+        }
+
+        attributes = LLMSpanAttributes(**span_attributes)
+
+        span = tracer.start_span(
+            "openai.completion",
+            kind=SpanKind.CLIENT,
+            context=set_span_in_context(trace.get_current_span()),
+        )
+        _set_input_attributes(span, kwargs, attributes)
+
+        try:
+            result = wrapped(*args, **kwargs)
+            if is_streaming(kwargs):
+                prompt_tokens = 0
+                for message in kwargs.get("messages", {}):
+                    prompt_tokens += calculate_prompt_tokens(
+                        json.dumps(str(message)), kwargs.get("model")
+                    )
+
+                if (
+                    kwargs.get("functions") is not None
+                    and kwargs.get("functions") != NOT_GIVEN
+                ):
+                    for function in kwargs.get("functions"):
+                        prompt_tokens += calculate_prompt_tokens(
+                            json.dumps(function), kwargs.get("model")
+                        )
+
+                return StreamWrapper(
+                    result,
+                    span,
+                    prompt_tokens,
+                    function_call=kwargs.get("functions") is not None,
+                    tool_calls=kwargs.get("tools") is not None,
+                )
+            else:
+                _set_response_attributes(span, kwargs, result)
+                span.set_status(StatusCode.OK)
+                span.end()
+                return result
+
+        except Exception as error:
+            span.record_exception(error)
```

Comment:
OTel instrumentations don't usually record exception - exceptions are huge and expensive and users are expected to log them themselves if they want to. It usually makes sense to record unhandled exceptions and here we don't know if it's handled.

Reply:
agree, will fix